runtime/src/main/java/com/flipkart/flux/guice/interceptor/TransactionInterceptor.java

```diff
@@ -11,41 +11,60 @@
  * limitations under the License.
  */
 
-package com.flipkart.flux.util;
+package com.flipkart.flux.guice.interceptor;
 
 import org.aopalliance.intercept.MethodInterceptor;
 import org.aopalliance.intercept.MethodInvocation;
+import org.hibernate.HibernateException;
 import org.hibernate.Session;
+import org.hibernate.SessionFactory;
 import org.hibernate.Transaction;
 import org.hibernate.context.internal.ManagedSessionContext;
+import javax.inject.Inject;
 
 /**
- * Provides transactional boundaries to methods which are annotated with {@link com.flipkart.flux.util.Transactional}.
+ * <code>TransactionInterceptor</code> is a {@link MethodInterceptor} implementation to provide
+ * transactional boundaries to methods which are annotated with {@link javax.transaction.Transactional}.
+ *
+ * Example:
+ * {
+ *     method1(); //call method1 which is annotated with transactional
+ * }
+ *
+ * @Transactional
+ * void method1() {
+ *      method2(); //call method2 which is annotated with transactional
+ * }
+ *
+ * @Transactional
+ * void method2() {}
+ *
+ * In the above case a transaction would be started before method1 invocation using this interceptor and ended once method1's execution is over.
+ * Same session and transaction would be used throughout.
+ *
  * @author shyam.akirala
  */
 public class TransactionInterceptor implements MethodInterceptor {
 
-    private ThreadLocal<Session> threadLocalSession;
-
-    public TransactionInterceptor() {
-        threadLocalSession = new ThreadLocal<Session>();
-    }
+    @Inject
+    private SessionFactory sessionFactory;
```

Comment:
Constructor injection is not possible here as the interceptor is getting created in Module's configure method manually.

TransactionInterceptor transactionInterceptor = new TransactionInterceptor();
requestInjection(transactionInterceptor);
bindInterceptor(Matchers.any(), Matchers.annotatedWith(Transactional.class), transactionInterceptor);